src/Symfony/Bridge/Twig/AppVariable.php

```diff
@@ -59,7 +73,11 @@ public function setDebug(bool $debug)
     public function getToken()
     {
         if (null === $tokenStorage = $this->tokenStorage) {
-            throw new \RuntimeException('The "app.token" variable is not available.');
+            if ($this->strictVariables) {
```

Comment:
You should also keep the previous condition, isn't it?
```suggestion
            if (null === $tokenStorage = $this->tokenStorage && $this->strictVariables) {
```
Same for `getUser()`, `getRequest()` and `getSession()`

Reply:
We need to throw Exception or return `null` only if `$this->tokenStorage !== null`, doing this we cannot return null, we need the two `if` separated (the second one to decide weither throwing exception or not)

Reply:
We may not do it for `getRequst` too, if `$this->requestStack === null && $this->strictVariables === false` we got to retrieve `$this->getRequest()` and do the same check again to finally return null, I suggest to keep it simple as it and keep the second if only to decide weither throwing exception or return null